### ISSUE TITLE
[FEAT] 라벨링 에러 처리를 위한 cleanlab 사용

### ISSUE THREAD

**Issue by lectura7942** [boostcampaitech5/level2_nlp_datacentric-nlp-07#21]
## Background

- g2p 제거한 데이터셋 사용

### cleanlab
- confident learning을 통해 레이블 오류를 예측한다.
> when the predicted probability of an example is **greater than a per-class-threshold**, we confidently count that example as actually belonging to that threshold’s class. The thresholds for each class are the **average predicted probability of examples** in that class.
> \- [An Introduction to Confident Learning: Finding and Learning with Label Errors in Datasets](https://l7.curtisnorthcutt.com/confident-learning)
- 각 레이블에 대한 예측 확률을 사용. 예측 확률을 생성할 모델은 제공해야 한다
- 라벨링 오류를 예측할 데이터는 학습 데이터에 속하지 않은 out-of-sample이야 하므로 cross-validation 사용

## 코드

```python
# data = 라벨 노이즈 있는 학습 데이터
# labels = 라벨 노이즈 있는 학습 데이터의 라벨
# pred_probs = kfold를 통해 valid set에 대한 예측 확률, size = (데이터 개수, 7)

# 라벨 에러로 탐지된 인덱스 배열 반환
issue_idx = cleanlab.filter.find_label_issues(labels, pred_probs, return_indices_ranked_by="self_confidence", 
                                              filter_by='prune_by_noise_rate') # default

display(data.iloc[issue_idx])
```

cleanlab 공식문서에서 다음과 같이 라벨 에러 탐지 기준을 제공합니다.

![image](https://github.com/boostcampaitech5/level2_nlp_datacentric-nlp-07/assets/81620001/712894b7-894a-4ee0-8284-d925fc86ac97)

## To Do
- kobert로 text embedding 추출 후 classifier만 파인튜닝한 모델 사용
  - [x] LogisticRegression (cleanlab 튜토리얼)
  - [ ] nn.Linear classifier (huggingface, 베이스코드에서 사용하는 구조) (코드를 잘못 짰는지 점수가 낮습니다)
- [ ] kobert 전체 파인튜닝한 모델 사용

## See Also
- [Confident Learning: Estimating Uncertainty in Dataset Labels](https://arxiv.org/abs/1911.00068)
- [An Introduction to Confident Learning: Finding and Learning with Label Errors in Datasets](https://l7.curtisnorthcutt.com/confident-learning)
